experimental/addons/hive-operator/README.md

```diff
@@ -0,0 +1,175 @@
+# hive-operator
+
+Apache Hive is a distributed, fault-tolerant data warehouse system that enables analytics at a massive scale. Hive Metastore(HMS) provides a central repository of metadata that can easily be analyzed to make informed, data driven decisions, and therefore it is a critical component of many data lake architectures. Hive is built on top of Apache Hadoop and supports storage on S3, adls, gs etc though hdfs. Hive allows users to read, write, and manage petabytes of data using SQL.
+
+This is an operator for Kubernetes that can manage Apache Hive. Currently, it only supports the Hive Metastore!
+
+## Install Operator
+
+Add experimental registry
+```
+vela addon registry add experimental --type=helm --endpoint=https://addons.kubevela.net/experimental/
+```
+
+Enable this addon
+```
+vela addon enable hive-operator
+```
+
+```shell
+$ vela ls -A | grep hive
+vela-system     addon-hive-operator     ns-hive-operator                        k8s-objects                                     running healthy
+vela-system     └─                      hive-operator                           helm                                            running healthy Fetch repository successfully, Create helm release
+```
+
+Disable this addon
+```
+vela addon disable hive-operator
+```
+
+## Install Dependencies
+
+In order to install the MinIO and PostgreSQL dependencies via Helm, you have to deploy two charts.
+
+**Minio**
+
+```shell
+helm install minio \
```

Comment:
1. can we use vela application or leverage addon for this installation?
2. the further question is: if I were the user, I have already learned `helm install`, why I don't use helm install to install hive?

Reply:
Extended addon capabililty to install dependencies at the time of addon installation.

Reply:
Sorry but I don't think this conversation is resolved, yet. What @wonderflow means is this addon needs to have its distinct advantages compared to helm, otherwise the user would choose helm instead. 

For example, vela Definitions provide an easy way to spin up complex things. In this particular addon, you have 3 components `hive-cluster`, `s3-connection` and `secret-class` that must be applied in order to have a functional hive cluster. Can this process be simplified? For example, can the user spin up a hive cluster with just one well-designed component? That well-designed component creates several resources under the hood, abstracts away tedious details from user while maintaining a certain level of customizability. And this is what makes vela or vela addons apart from typical helm installation.

These are just my thoughts on this issue.

Reply:
Thanks @charlie0129 for explanation. And it makes sence to have a simplified installation process. I think I should create a new component that create all hive related resources at once and make hive-cluster running in the cluster.